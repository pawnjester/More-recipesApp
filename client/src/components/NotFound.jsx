import React from 'react';
import NavigationBar from './NavigationBar';

const NotFound = () =>
  (<div>
    <NavigationBar search="true" />
    <div className="container">
      <div className="row mt-5">
        <div className="col-md-4">
          <i className="fa fa-3x fa-exclamation-circle " style={{ fontSize: 300 }} /><br />
        </div>
        <div className="col-md-8">
          <h3 style={{ fontSize: 40 }}>Page not found</h3><br />
          <p>We are sorry but the page you are looking for does not exist.</p>
        </div>
      </div>
    </div>
  </div>
  );

export default NotFound;
