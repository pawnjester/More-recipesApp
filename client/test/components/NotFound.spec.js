import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import expect from 'expect';
import NotFound from '../../src/components/NotFound';
import store from '../../src/store/store';


configure({ adapter: new Adapter() });

const setup = () => {
  const shallowWrapper = shallow(<NotFound />);
  const mountedWrapper = mount(
    <Provider store={store}>
      <BrowserRouter>
        <NotFound store={store} />
      </BrowserRouter>
    </Provider>);
  return {
    shallowWrapper,
    mountedWrapper
  };
};
describe('Test For Not Found', () => {
  it('should render correctly', () => {
    const { mountedWrapper } = setup();
    expect(mountedWrapper).toMatchSnapshot();
  });
});
