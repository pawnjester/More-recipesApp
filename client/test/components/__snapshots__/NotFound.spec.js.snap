// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test For Not Found should render correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "blank",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <NotFound
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
      >
        <div>
          <Connect(NavigationBar)
            search="true"
          >
            <NavigationBar
              auth={
                Object {
                  "deleted": false,
                  "error": Object {},
                  "isAuthenticated": false,
                  "mostFavorites": Object {},
                  "pages": 1,
                  "recipes": Array [],
                  "searchResults": Array [],
                  "totalContent": 1,
                  "upvotedRecipes": Array [],
                  "user": Object {},
                }
              }
              getUserDetail={[Function]}
              logout={[Function]}
              search="true"
              userDetail={Object {}}
            >
              <nav
                className="navbar navbar-expand-sm navbar-dark navbar-inverse bg-dark fixed-top "
              >
                <Link
                  className="navbar-brand"
                  replace={false}
                  styles="cursor:pointer"
                  to="/"
                >
                  <a
                    className="navbar-brand"
                    href="/"
                    onClick={[Function]}
                    styles="cursor:pointer"
                  >
                    More-Recipes
                  </a>
                </Link>
                <Connect(SearchBar)>
                  <SearchBar
                    searchRecipe={[Function]}
                  >
                    <div>
                      <form
                        className=" my-2 my-lg-0"
                      >
                        <input
                          aria-label="Search"
                          className=" mr-sm-2 searchbars"
                          id="search-bar"
                          name="search"
                          onChange={[Function]}
                          onFocus={[Function]}
                          placeholder="Search recipes...."
                          style={
                            Object {
                              "width": 320,
                            }
                          }
                          type="text"
                          value=""
                        />
                      </form>
                    </div>
                  </SearchBar>
                </Connect(SearchBar)>
                <button
                  aria-controls="navbarSupportedContent"
                  aria-expanded="false"
                  aria-label="Toggle navigation"
                  className="navbar-toggler"
                  data-target="#navbarSupportedContent"
                  data-toggle="collapse"
                  type="button"
                >
                  <span
                    className="navbar-toggler-icon"
                  />
                </button>
                <div
                  className="collapse navbar-collapse"
                  id="navbarSupportedContent"
                >
                  <ul
                    className="navbar-nav ml-auto"
                  >
                    <Link
                      className="nav-link nov bg-dark"
                      id="sign-in-link"
                      replace={false}
                      styles="border-radius: 4px;"
                      to="/signin"
                    >
                      <a
                        className="nav-link nov bg-dark"
                        href="/signin"
                        id="sign-in-link"
                        onClick={[Function]}
                        styles="border-radius: 4px;"
                      >
                        Sign In
                      </a>
                    </Link>
                  </ul>
                </div>
              </nav>
            </NavigationBar>
          </Connect(NavigationBar)>
          <div
            className="container"
          >
            <div
              className="row mt-5"
            >
              <div
                className="col-md-4"
              >
                <i
                  className="fa fa-3x fa-exclamation-circle "
                  style={
                    Object {
                      "fontSize": 300,
                    }
                  }
                />
                <br />
              </div>
              <div
                className="col-md-8"
              >
                <h3
                  style={
                    Object {
                      "fontSize": 40,
                    }
                  }
                >
                  Page not found
                </h3>
                <br />
                <p>
                  We are sorry but the page you are looking for does not exist.
                </p>
              </div>
            </div>
          </div>
        </div>
      </NotFound>
    </Router>
  </BrowserRouter>
</Provider>
`;
